src/function-builder.ts

```diff
@@ -44,45 +52,79 @@ import * as storage from './providers/storage';
- * @throws { Error } Memory and TimeoutSeconds values must be valid.
+ * @throws { Error } FailurePolicy, Memory and TimeoutSeconds values must be
+ * valid.
  */
-function assertRuntimeOptionsValid(runtimeOptions: RuntimeOptions): boolean {
-  if (
-    runtimeOptions.memory &&
-    !_.includes(VALID_MEMORY_OPTIONS, runtimeOptions.memory)
-  ) {
-    throw new Error(
-      `The only valid memory allocation values are: ${VALID_MEMORY_OPTIONS.join(
-        ', '
-      )}`
-    );
+function assertRuntimeOptionsValidity(runtimeOptions: RuntimeOptions): void {
+  if (isObjectLike(runtimeOptions) === false) {
+    throw new Error('RuntimeOptions must be an object.');
   }
-  if (
-    runtimeOptions.timeoutSeconds > MAX_TIMEOUT_SECONDS ||
-    runtimeOptions.timeoutSeconds < 0
-  ) {
-    throw new Error(
-      `TimeoutSeconds must be between 0 and ${MAX_TIMEOUT_SECONDS}`
-    );
+
+  const { failurePolicy, memory, timeoutSeconds } = runtimeOptions;
+
+  if (failurePolicy !== undefined) {
+    if (
+      isBoolean(failurePolicy) === false &&
+      isObjectLike(failurePolicy) === false
+    ) {
+      throw new Error(
+        `RuntimeOptions.failurePolicy must be a boolean or an object.`
+      );
+    }
+
+    if (typeof failurePolicy === 'object') {
+      if (
+        isObjectLike(failurePolicy.retry) === false ||
+        isEmpty(failurePolicy.retry) === false
+      ) {
+        throw new Error(
+          'RuntimeOptions.failurePolicy.retry must be an empty object.'
+        );
+      }
+    }
+  }
+
+  if (memory !== undefined) {
+    if (includes(VALID_MEMORY_OPTIONS, memory) === false) {
+      throw new Error(
+        `RuntimeOptions.memory must be one of: ${VALID_MEMORY_OPTIONS.join(
+          ', '
+        )}.`
+      );
+    }
+  }
+
+  if (timeoutSeconds !== undefined) {
+    if (typeof timeoutSeconds !== 'number') {
+      throw new Error('RuntimeOptions.timeoutSeconds must be a number.');
+    }
+
+    if (
+      timeoutSeconds < MIN_TIMEOUT_SECONDS ||
+      timeoutSeconds > MAX_TIMEOUT_SECONDS
+    ) {
+      throw new Error(
+        `RuntimeOptions.timeoutSeconds must be between ${MIN_TIMEOUT_SECONDS} and ${MAX_TIMEOUT_SECONDS}.`
+      );
+    }
   }
-  return true;
 }
 
 /**
  * Assert regions specified are valid.
  * @param regions list of regions.
  * @throws { Error } Regions must be in list of supported regions.
  */
-function assertRegionsAreValid(regions: string[]): boolean {
-  if (!regions.length) {
-    throw new Error('You must specify at least one region');
+function assertRegionsValidity(regions: string[]): void {
+  if (regions.length === 0) {
+    throw new Error('You must specify at least one region.');
   }
-  if (_.difference(regions, SUPPORTED_REGIONS).length) {
+
+  if (difference(regions, SUPPORTED_REGIONS).length !== 0) {
     throw new Error(
-      `The only valid regions are: ${SUPPORTED_REGIONS.join(', ')}`
+      `The only valid regions are: ${SUPPORTED_REGIONS.join(', ')},`
```

Comment:
Should that be a period instead of a comma at the end there?

Reply:
Thanks, fixed in 1c42e0ec766d13351072d560d82b799f08311ec1.